pytools/graph.py

```diff
@@ -420,4 +421,45 @@ def validate_graph(graph: Mapping[T, Collection[T]]) -> None:
 
 # }}}
 
+
+# {{{
+
+def is_connected(graph: Mapping[T, Collection[T]]) -> bool:
+    """
+    Returns whether all nodes in *graph* are connected, ignoring
+    the edge direction.
+
+    :arg graph: A :class:`collections.abc.Mapping` representing a directed
+        graph. The dictionary contains one key representing each node in the
+        graph, and this key maps to a :class:`collections.abc.Collection` of nodes
+        that are connected to the node by outgoing edges.
+    """
+    validate_graph(graph)
```

Comment:
I don't love this: it imposes cost on the user even when they know that their graph is valid.

Reply:
Good point, I removed this in 99fd92b